Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/Home.kt

```diff
@@ -67,107 +64,139 @@ import androidx.compose.ui.unit.Dp
-            )
-        }
-    ) { innerPadding ->
-        val modifier = Modifier.padding(innerPadding)
-        Crossfade(currentSection) { section ->
-            when (section) {
-                HomeSections.Feed -> Feed(
-                    onSnackClick = onSnackSelected,
-                    modifier = modifier
-                )
-                HomeSections.Search -> Search(onSnackSelected, modifier)
-                HomeSections.Cart -> Cart(onSnackSelected, modifier)
-                HomeSections.Profile -> Profile(modifier)
-            }
-        }
+    composable(HomeSections.SEARCH.route) { from ->
+        Search(onSnackClick = { id -> onSnackSelected(id, from) }, modifier)
+    }
+    composable(HomeSections.CART.route) { from ->
+        Cart(onSnackClick = { id -> onSnackSelected(id, from) }, modifier)
     }
+    composable(HomeSections.PROFILE.route) {
+        Profile(modifier)
+    }
+}
+
+private object HomeDestinations {
+    const val FEED_ROUTE = "home/feed"
+    const val SEARCH_ROUTE = "home/search"
+    const val CART_ROUTE = "home/cart"
+    const val PROFILE_ROUTE = "home/profile"
+}
+
+enum class HomeSections(
+    @StringRes val title: Int,
+    val icon: ImageVector,
+    val route: String
+) {
+    FEED(R.string.home_feed, Icons.Outlined.Home, HomeDestinations.FEED_ROUTE),
+    SEARCH(R.string.home_search, Icons.Outlined.Search, HomeDestinations.SEARCH_ROUTE),
+    CART(R.string.home_cart, Icons.Outlined.ShoppingCart, HomeDestinations.CART_ROUTE),
+    PROFILE(R.string.home_profile, Icons.Outlined.AccountCircle, HomeDestinations.PROFILE_ROUTE)
 }
 
 @Composable
-private fun JetsnackBottomNav(
-    currentSection: HomeSections,
-    onSectionSelected: (HomeSections) -> Unit,
-    items: List<HomeSections>,
+fun JetsnackBottomBar(
+    navController: NavController,
+    tabs: Array<HomeSections>,
     color: Color = JetsnackTheme.colors.iconPrimary,
     contentColor: Color = JetsnackTheme.colors.iconInteractive
 ) {
-    JetsnackSurface(
-        color = color,
-        contentColor = contentColor
-    ) {
-        val springSpec = remember {
-            SpringSpec<Float>(
-                // Determined experimentally
-                stiffness = 800f,
-                dampingRatio = 0.8f
-            )
-        }
-        JetsnackBottomNavLayout(
-            selectedIndex = currentSection.ordinal,
-            itemCount = items.size,
-            indicator = { JetsnackBottomNavIndicator() },
-            animSpec = springSpec,
-            modifier = Modifier.navigationBarsPadding(left = false, right = false)
+
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    val currentRoute = navBackStackEntry?.arguments?.getString(KEY_ROUTE)
+        ?: HomeSections.FEED.route
+
+    val routes = remember { HomeSections.values().map { it.route } }
+    if (currentRoute in routes) {
+        val currentSection = HomeSections.values().first { it.route == currentRoute }
```

Comment:
`HomeSections.values()` will create a new array every composition. Cache the value somewhere (could be a top level val)

Reply:
Fixed